mqtt-examples/src/main/java/io/vertx/example/mqtt/simple/Client.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2016 Red Hat Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.vertx.example.mqtt.simple;
+
+import io.netty.handler.codec.mqtt.MqttQoS;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.example.mqtt.util.Runner;
+import io.vertx.mqtt.MqttClient;
+import io.vertx.mqtt.MqttClientOptions;
+
+/**
+ * An example of using the MQTT client as a verticle
+ */
+public class Client extends AbstractVerticle {
+
+  private static final String MQTT_TOPIC = "/my_topic";
+  private static final String MQTT_MESSAGE = "Hello Vert.x MQTT Client";
+
+  // Convenience method so you can run it in your IDE
+  public static void main(String[] args) {
+    Runner.runExample(Client.class);
+  }
+
+  @Override
+  public void start() throws Exception {
+    MqttClientOptions options = new MqttClientOptions()
+      .setPort(1883)
+      .setHost("0.0.0.0");
+
+    MqttClient mqttClient = MqttClient.create(vertx, options);
+
+    mqttClient.connect(ch -> {
+      if(ch.succeeded()) {
+        System.out.println("Connected to a server");
+
+        mqttClient.publish(
+          MQTT_TOPIC,
+          Buffer.buffer(MQTT_MESSAGE.getBytes()),
+          MqttQoS.AT_MOST_ONCE,
+          false,
+          false,
+          s -> System.out.println("Publish sent to a server"));
```

Comment:
This example just connects and publishes without any clean disconnection maybe it's better showing clean disconnection as all MQTT clients should do (after publishing).